File: swig/python/test-r_asm.py
from r_asm import *

def disasm(a, arch, op):
	print "---------------------------->8- - - - - -"
	print "OPCODE: %s"%op
	a.use (arch)
	print "ARCH: %s"%arch
	code = a.massemble (op)
	if code is None:
		print "HEX: Cannot assemble opcode"
	else:
		print "HEX: %s"%code.buf_hex
	
a = rAsm()
print "---[ name ]-----[ description ]----------"
a.list()

disasm (a, 'x86.olly', 'mov eax, 33')
disasm (a, 'java', 'bipush 33')
disasm (a, 'java', 'invalid opcode')

File: swig/perl/test-r_asm.pl
#!/usr/bin/perl
require "r_asm.pm";

sub disasm {
	my ($a, $arch, $op) = @_;
	print("---\n");
	print("OPCODE: $op\n");
	$a->use ($arch);
	print("ARCH: $arch\n");
	my $code = $a->massemble ($op);
	if (defined($code)) {
		my $buf = r_asmc::rAsmCode_buf_hex_get ($code);
		print "HEX: $buf\n";
	} else {
		print("HEX: Cannot assemble opcode\n");
	}
}

my $a = new r_asm::rAsm();
$a->list();

disasm ($a, 'x86.olly', 'mov eax, 33');
disasm ($a, 'java', 'bipush 33');
disasm ($a, 'java', 'invalid opcode');
